src/components/MarkdownEditor.js

```diff
@@ -19,47 +38,63 @@ class MarkdownEditor extends React.Component {
     super(props);
     this.handleChange = this.handleChange.bind(this);
     this.state = {
-      reactMdeValue: { text: props.defaultValue, selection: null },
+      value: props.defaultValue || '',
     };
+    this.converter = new Showdown.Converter();
+    this.commands = [
+      {
+        commands: [
+          Commands.headerCommand,
+          Commands.boldCommand,
+          Commands.italicCommand,
+        ],
+      },
+      {
+        commands: [Commands.linkCommand, Commands.imageCommand],
+      },
+      {
+        commands: [Commands.unorderedListCommand, Commands.orderedListCommand],
+      },
+    ];
   }
 
   handleChange(value) {
-    this.setState({ reactMdeValue: value });
-    this.props.onChange(value.text);
+    this.setState({ value });
+    this.props.onChange(value);
   }
 
   render() {
     return (
       <div
-        className={`${
-          this.props.preview === false ? 'noPreview' : ''
-        } MarkdownEditor`}
+        className={classNames('MarkdownEditor', {
+          noPreview: this.props.preview === false,
+        })}
       >
         <style jsx global>
           {`
-            .MarkdownEditor .react-mde-editor {
-              width: 50%;
-            }
-            .MarkdownEditor.noPreview .react-mde-preview {
+            .MarkdownEditor.noPreview .mde-tabs {
               display: none;
             }
-            .MarkdownEditor .react-mde-preview {
-              width: 50%;
-            }
-            .MarkdownEditor .column .mde-preview .mde-preview-content {
-              padding-top: 5px;
+            .DraftEditor-root,
+            .DraftEditor-editorContainer,
+            .public-DraftEditor-content {
+              height: 100%;
             }
           `}
         </style>
         <ReactMde
-          textAreaProps={{
-            id: 'ta1',
-            name: 'ta1',
-          }}
-          value={this.state.reactMdeValue}
-          className="column"
+          value={this.state.value}
           onChange={this.handleChange}
-          commands={ReactMdeCommands.getDefaultCommands()}
+          commands={this.commands}
+          generateMarkdownPreview={markdown => {
+            return Promise.resolve(this.converter.makeHtml(markdown));
```

Comment:
Why do we need Promise.resolve here?

Reply:
react-mde requires a promise to be returned for `generateMarkdownPreview`. I've added a comment for this.